contracts/SimpleGAMultiSig.aes

```diff
@@ -42,120 +42,171 @@ contract SimpleGAMultiSig =
-      Some(ga_tx_hash) =>
+      Some(tx_hash) =>
         require(has_consensus(tx), "ERROR_NO_CONSENSUS")
-        require(tx.ga_tx_hash == ga_tx_hash, "ERROR_UNEQUAL_HASHES")
-        Chain.event(TxAuthorized(ga_tx_hash))
-        put(state{ nonce = nonce + 1 })
-        put(state{ current_tx = None })
+        require(tx.tx_hash == tx_hash, "ERROR_UNEQUAL_HASHES")
+        Chain.event(TxAuthorized(tx_hash))
+        put(state{ nonce = nonce + 1, current_tx = None })
         true
 
   // propose a new tx valid for the given ttl
-  stateful entrypoint propose(ga_tx_hash : hash, ttl : Chain.ttl) =
+  stateful entrypoint propose(tx_hash : hash, ttl : Chain.ttl) =
     require(Set.member(Call.caller, state.signers), "ERROR_NOT_AUTHORIZED")
     switch(state.current_tx)
-      None => set_current_tx(ga_tx_hash, ttl)
+      None => set_current_tx(tx_hash, ttl)
       Some(tx) =>
-        require(!not_expired(tx), "ERROR_EXISTING_PROPOSED_TX_NOT_EXPIRED")
-        set_current_tx(ga_tx_hash, ttl)
+        require(expired(tx), "ERROR_EXISTING_PROPOSED_TX_NOT_EXPIRED")
+        set_current_tx(tx_hash, ttl)
 
   // signer confirms the current tx
-  stateful entrypoint confirm(ga_tx_hash : hash) =
+  stateful entrypoint confirm(tx_hash : hash) =
     let tx = Option.force_msg(state.current_tx, "ERROR_NOTHING_TO_CONFIRM")
-    require_valid_signer_and_tx_hash(Call.caller, ga_tx_hash, tx)
+    require_valid_signer_and_tx_hash(Call.caller, tx_hash, tx)
     require(!Set.member(Call.caller, tx.confirmed_by), "ERROR_ALREADY_CONFIRMED")
-    require(not_expired(tx), "ERROR_TX_EXPIRED")
+    require(!expired(tx), "ERROR_TX_EXPIRED")
     let updated_tx = tx{ confirmed_by = Set.insert(Call.caller, tx.confirmed_by) }
     put(state{ current_tx = Some(updated_tx) })
-    Chain.event(TxConfirmed(tx.ga_tx_hash, Call.caller))
-    if(has_consensus(updated_tx))
-      Chain.event(TxConsensusReached(tx.ga_tx_hash))
+    Chain.event(TxConfirmed(tx.tx_hash, Call.caller))
+    if(state.confirmations_required == Set.size(updated_tx.confirmed_by))
+      Chain.event(TxConsensusReached(tx.tx_hash))
 
   // revoke the current tx and clean state
-  stateful entrypoint revoke(ga_tx_hash : hash) =
+  stateful entrypoint revoke(tx_hash : hash) =
     let tx = Option.force_msg(state.current_tx, "ERROR_NOTHING_TO_REVOKE")
-    require_valid_signer_and_tx_hash(Call.caller, ga_tx_hash, tx)
+    require_valid_signer_and_tx_hash(Call.caller, tx_hash, tx)
     put(state{ current_tx = None })
-    Chain.event(TxRevoked(tx.ga_tx_hash, Call.caller))
+    Chain.event(TxRevoked(tx.tx_hash, Call.caller))
 
-  // return all signers
-  entrypoint get_signers() : list(address) =
-  	Set.to_list(state.signers)
+  stateful entrypoint update_fee_protection(fee_protection : fee_protection) =
+    require(Set.member(Call.caller, state.signers), "ERROR_NOT_A_VALID_SIGNER")
+    require(fee_protection.max_fee >= 2_000_000_000_000_000, "ERROR_MAX_FEE_MUST_NOT_BE_DECREASED")
```

Comment:
That check I don't understand I could think of cases (during a hype) when you'd like to raise the `max_fee` and later lower it again 🤔

Reply:
@hanssv I updated the error message. I think we could discuss lower default values. but IMO it shouldn't be possible to allow lower values than those we set by default. I agree it should be able to increase and decrease based on network workload (e.g. during a hype)